lib/broadway_dashboard.ex

```diff
@@ -111,12 +143,19 @@ defmodule BroadwayDashboard do
 
   @impl true
   def render_page(assigns) do
-    items =
-      for name <- assigns.pipelines do
-        {name, name: format_nav_name(name), render: render_pipeline(assigns), method: :redirect}
-      end
+    if assigns.pipelines == [] do
+      render_error(assigns)
+    else
+      items =
+        for name <- assigns.pipelines do
+          {name,
+           name: format_nav_name(name),
+           render: render_pipeline_or_error(assigns),
```

Comment:
Can we make the `render` function an anonymous function so we don't render all pipelines when we need to show only one of them? 🤔

Reply:
We should probably enforce it to be an anonymous function in the dashboard...

Reply:
> We should probably enforce it to be an anonymous function in the dashboard...

Yeah, I think this can avoid unnecessary renders.